test/metabase/api/embed_test.clj

```diff
@@ -670,6 +684,25 @@
             (is (= 101
                    (count (csv/read-csv results))))))))))
 
+(deftest embed-download-query-execution-test
+  (testing "Tests that embedding download context shows up in the query execution table when downloading cards."
+    ;; Clear out the query execution log so that test doesn't read stale state
+    (t2/delete! :model/QueryExecution)
+    (mt/with-ensure-with-temp-no-transaction!
+      (with-embedding-enabled-and-new-secret-key
+        (with-temp-dashcard [dashcard {:dash {:enable_embedding true}
+                                       :card {:dataset_query (mt/mbql-query venues)}}]
+          (client/client :get 200 (str (dashcard-url dashcard) "/csv"))
+          ;; the query execution is saved async, so we need to sleep a bit
```

Comment:
copied from https://github.com/metabase/metabase/blob/jh-track-embed/test/metabase/actions/execution_test.clj#L19

Reply:
Can you use `metabase.query-processor.middleware.process-userland-query-test/with-query-execution` here instead? It redefs `save-query-execution*` so that it stores the data in a promise, and then derefs it with a timeout instead of waiting a constant 200 ms each time. Bonus points if you can switch the actions test to use it as well.

Sleeping for hardcoded intervals in tests is generally an antipattern. A single case isn't very noticeable but if it proliferates it can add a lot of waiting time to the test suite.